docs/telemetry.md

```diff
@@ -1,3 +1,156 @@
+    }
+}
+```
+<!-- endSnippet -->
+
+Alternatively, you can use the `AddResiliencePipeline(...)` extension method which automatically enables telemetry for defined pipeline:
+
+<!-- snippet: add-resilience-pipeline-with-telemetry -->
+```cs
+var serviceCollection = new ServiceCollection()
+    .AddLogging(builder => builder.AddConsole())
+    .AddResiliencePipeline("my-strategy", builder => builder.AddTimeout(TimeSpan.FromSeconds(1)))
+    .Configure<TelemetryOptions>(options =>
+    {
+        // Configure enrichers
+        options.MeteringEnrichers.Add(new MyMeteringEnricher());
+
+        // Configure telemetry listeners
+        options.TelemetryListeners.Add(new MyTelemetryListener());
+    });
+```
+<!-- endSnippet -->
+
+## Metrics
+
+The emitted metrics are emitted under the `Polly` meter name. The subsequent sections provide insights into the metrics produced by Polly. Please note that any custom enriched dimensions are not depicted in the following tables.
+
+### resilience-events
+
+- Type: *Counter*
+- Description: Emitted upon the occurrence of a resilience event.
+
+Dimensions:
+
+|Name|Description|
+|---| ---|
+|`event-name`| The name of the emitted event.|
+|`event-severity`| The severity of the event (`Debug`, `Information`, `Warning`, `Error`, `Critical`).|
+|`pipeline-name`| The name of the pipeline corresponding to the resilience pipeline.|
+|`pipeline-instance`| The instance name of the pipeline corresponding to the resilience pipeline.|
+|`strategy-name`| The name of the strategy generating this event.|
+|`operation-key`| The operation key associated with the call site. |
+|`exception-name`| The full name of the exception assigned to the execution result (`System.InvalidOperationException`). |
+
+### execution-attempt-duration
+
+- Type: *Histogram*
+- Unit: *milliseconds*
+- Description: Tracks the duration of execution attempts, produced by `Retry` and `Hedging` resilience strategies.
+
+Dimensions:
+
+|Name|Description|
+|---| ---|
+|`event-name`| The name of the emitted event.|
+|`event-severity`| The severity of the event (`Debug`, `Information`, `Warning`, `Error`, `Critical`).|
+|`pipeline-name`| The name of the pipeline corresponding to the resilience pipeline.|
+|`pipeline-instance`| The instance name of the pipeline corresponding to the resilience pipeline.|
+|`strategy-name`| The name of the strategy generating this event.|
+|`operation-key`| The operation key associated with the call site. |
+|`exception-name`| The full name of the exception assigned to the execution result (`System.InvalidOperationException`). |
+|`attempt-number`| The execution attempt number, starting at 0 (0, 1, 2). |
+|`attempt-handled`| Indicates if the execution outcome was handled. A handled outcome indicates execution failure and the need for retry (`true`, `false`). |
+
+### pipeline-execution-duration
+
+- Type: *Histogram*
+- Unit: *milliseconds*
+- Description: Measures the duration and results of resilience pipelines.
```

Comment:
It doesn't contain the result anymore does it?

Reply:
True.